### ISSUE TITLE
Revise UTD to not be Unnecessary when it gives additional positive information

### ISSUE THREAD

**Issue by pianoblook** [hanabi/hanabi.github.io#624]
(opening this issue in response to Valetta's question on Discord)

This just seems like a no-brainer, and I assumed this was how we all played already. 

- We already agree that a UTD shouldn't be considered Unnecessary if it gives additional positive information to the clue receiver's hand. 
- In other words, we agree that a UTD is to be considered inherently valuable + efficient when it's able to provide more positive information.
- Therefore, there's zero reason why this shouldn't extend to UTDs that give a **different** teammate new + exact positive ID on a previously clued card.

**Comment by Zamiell** [comment]
> zero reason

how about this:
superpositions are (sometimes) asymmetric.
in other words, it is not always possible for bob to determine that alice's clue reduces an arbitrary superposition.
on the other hand, reducing superpositions in the clue receiver's hand is always completely symmetric.
in my mind, that is the primary reason for the convention currently being configured in the way that it is.

**Comment by pianoblook** [comment]
All this would change, btw, is the last line of the Unnecessary [entry](https://hanabi.github.io/docs/level-23/#unnecessary-moves):

> A move is not unnecessary if it also added more positive information to other previously-clued cards.

The only change is deleting the last 3 words, "...in the hand"

**Comment by pianoblook** [comment]
> superpositions are (sometimes) asymmetric.
> in other words, it is not always possible for bob to determine that alice's clue reduces an arbitrary superposition.

I don't follow. Are you saying Bob shouldn't Discharge?

- Once he Discharges, he can know with 100% certainty that everyone on the the team will have marked that card as Known Trash.

- So in the case where it was trash only because of it being a dupe of a clued but yet-to-be positively ID'd card, Bob instantly knows with 100% certainty that it has now become positively ID'd.

**Comment by florian-5f3759df** [comment]
I agree that that giving extra information in other clued hands is a great bonus of UTDs, but the annoying part is that this knowledge not global in some situations. So where would you draw the line? How to formulate this in a clean way?

For example: If https://hanab.live/replay/479057#6 happens, it fully identifies y3 after jack's discard. But imagine the same situation but with sin_faith having a globally known b3 clued. Then jack doesn't know yet whether they are chop moved or not, and can only identify clue focus on their own hand after discarding the trash 3. (Chop move if it's b3, no chop move if it's something else.)

This is a type of non-global information that can be a bit annoying. The same holds for https://hanab.live/replay/208817#5. In that case it's very clean, but it can easily be the case that someone already knows their card (say because they see two r4s and r5 in someone else's hand), but the fact that they already know their card is not known by the player holding r5.

**Comment by pianoblook** [comment]
That problem can exist regardless of who's receiving the clue though, right? For example:

All 1s are down
- Cathy holds a red card, but doesn't know the identity
- Alice clues red to Cathy, *newly* touching a card on slot 1
- Bob Discharges

At that precise moment, Cathy hasn't gained any new positive information - does that mean she needs to CM?

I'd strongly argue that no she should not, at least **UNLESS* she discards the known trash and finds out that it is red 1. 

If it turns out to be a red 3 though, for example, then retroactively she she trusts that she should not CM because, as the doc explains, "A move is not unnecessary if it also added more positive information to other previously-clued cards in the hand."

**Comment by Zamiell** [comment]
> Are you saying Bob shouldn't Discharge?

i think you misunderstand me

consider this replay: https://hanab.live/replay/208817#5

kim's red card is in a superposition of N states.
after padi plays i1, the superposition collapses.
in this particular game, this collapse *appears* to be symmetrical, at least at first glance.

however, this concept does not generalize to all games and it would be dangerous to assume that it does.
any arbitrary superposition is not guaranteed to be symmetrical.
in fact, it likely won't be symmetrical >=50% of the time.

for example, say that:
- bob has a superposition on a red card as [r3, r5] (r2 is played, Bob sees both 4's)
- cathy has a superposition on the same card as [r3, r4, r5]
- alice gives a clue that could either be *Necessarily* or *Unnecessary* 
- the clue eliminates r4 from the superposition on bob's hand
- from bob's perspective, the clue is *Unnecessary*, because it did not reduce the amount of cards in his superposition
- from cathy's perspective, the clue is *Necessary*, because it does reduce the amount of cards in Bob's superposition

**Comment by Zamiell** [comment]
i acknowledge that this problem can still occur when reducing a superposition in the clue receiver's hand.
however, since explicit information is being given, it seems less likely to occur, and the superposition modification is much more clear to the entire team.

**Comment by pianoblook** [comment]
So I see two ways forward:

1) Restrict the convention further to always CM unless **new information is actively filled in immediately** e.g. 
> b3 and r1 are played. Cathy holds one touched red card.
> Alice clues red to Cathy, touching a new card on slot 1 AND filling in her r3. 
> Bob Discharges.
> The clue itself has not immediately "added more positive information to other previously-clued cards", so Cathy does indeed CM, **regardless of what the slot 1 Known Trash ends up being**

2) Expand the convention to include positive information that is gained in other hands, even if not **immediately** realized.

If we reject (1), and instead agree that Cathy should *NOT* instantly CM in this example regardless of what her slot 1 card ends up being, I believe it would be extremely weird to not adopt (2)

**Comment by winsonnhm** [comment]
I am more concern on how people should react to clue focus if they don't have a chance to discard the known trash

In a No Variant 5 players game, max level

All 1s are down, Emily has a r4 on #4 clued as red

1. Alice clued red to Donald, touching only #2
2. Bob discharge, demonstrating a UTD, saying the red card is a trash
3. Cathy now clues yellow to Donald, touching #1 and #4, there is no yellow card on anyone's finesse position
4. According to the current argument, Donald is not certain if he is chop moved or not

- Donald is chop moved if the red card is r4, since it tells Emily her red card is exactly r4
- Donald is not chop moved if the red card is r1, since it gives no information to the team

Then in this case, is Donald suppose to defer the yellow play clue and discard the red known trash first? Will it looks like a scream from Emily's POV?

**Comment by Zamiell** [comment]
> Restrict the convention further to always CM unless new information is actively filled in immediately

this seems synonymous with what is currently written in the doc:

> A move is not unnecessary if it also added more positive information to other previously-clued cards in the hand.

**Comment by pianoblook** [comment]
> alice gives a clue that could either be Necessarily or Unnecessary
> the clue eliminates r4 from the superposition on bob's hand

I'm still really not getting this. If Bob is the one holding the red card, how can he also be the one Discharging?

**Comment by Zamiell** [comment]
> I'm still really not getting this. If Bob is the one holding the red card, how can he also be the one Discharging?

in this mockup i ignored the fact that bob was discharging.
change bob to cathy and change cathy to donald

**Comment by pianoblook** [comment]
> this seems synonymous with what is currently written in the doc:

To be very clear, you're saying Cathy *should* CM in the following example?:

> All 1s played
> Cathy's hand is r4, x, x, [r]4
> Alice clues red to Cathy and Bob _Discharges_
> Cathy discards the slot 1 card and finds it's r4. 
> The conventions dictate that **irregardless** of the fact that she now knows her other red card is exactly r4, she should still CM

Is that right?

**Comment by pianoblook** [comment]
> > I'm still really not getting this. If Bob is the one holding the red card, how can he also be the one Discharging?
> 
> in this mockup i ignored the fact that bob was discharging.
> to get around this, just change bob to cathy and change cathy to donald

who's being clued? Did Alice clue someone red? I'm not following your example. what are the hands?

**Comment by Zamiell** [comment]
> you're saying Cathy should CM in the following example?:

in this example, a superposition was reduced in the clue receiver's hand, which makes the clue useful, which therefore makes it *Necessary*, which therefore makes it not CM anything

**Comment by pianoblook** [comment]
Okay then I'm completely baffled and confused. You seem to have just contradicted yourself.

In response to my hypothetical proposal of, "Restrict the convention further to always CM unless new information is actively filled in immediately"

You responded: "this seems synonymous with what is currently written in the doc:"
> A move is not unnecessary if it also added more positive information to other previously-clued cards in the hand.

---- 
But just now, you said:
"in this example, a superposition was reduced in the clue receiver's hand, which makes the clue useful, which therefore makes it Necessary, which therefore makes it not CM anything"

So which is it:
(A) "synonymous with the doc" that it should **always CM unless new information is actively filled in immediately**
or 
(B) "Necessary, which therefore makes it not CM anything"

**Comment by Zamiell** [comment]
> who's being clued? Did Alice clue someone red? 

sure, alice clues emily red, or something.
i think the specific clue doesn't really matter though, since the concept could apply to any clue.
take the previous example where the superposition on kim's red 3 is collapsed.
if we were to give that move a name, we would call it a *Necessary UTD That Reduces a Non-Clue Receiver's Superposition*.

if:
- Alice performs an *Necessary UTD That Reduces a Non-Clue Receiver's Superposition*
- and Bob performs the *Discharge*
- and the *Clue Receiver* is Emily
- and the *Non-Clue Receiver with the Superposition* is Donald
- then throughout the team (Alice/Bob/Cathy/Donald/Emily) all might have different ideas as to whether or not Donald's superposition actually got reduced or not (because all 5 of them have different notes on the card)
- thus, some people on the team think that it is a *Necessary Move*, and other people on the team think that it is an*Unnecessary Move*

**Comment by Zamiell** [comment]
> (A) "synonymous with the doc" that it should always CM unless new information is actively filled in immediately
or
(B) "Necessary, which therefore makes it not CM anything"

i dont understand the difference here between these two options.
if a superposition is reduced, then the clue is *Necessary*, and it doesn't chop move.
if a superposition is not reduced (and the Discharge target wasn't blocked), then the clue is *Unnecessary*, and it does chop move

**Comment by pianoblook** [comment]
> throughout the team (Alice/Bob/Cathy/Donald/Emily) all might have different ideas as to whether or not Donald's superposition actually got reduced or not (because all 5 of them have different notes on Donald's card)
> * thus, some people on the team think that it is a _Necessary Move_, and other people on the team think that it is an _Unnecessary Move_

To me it just seems ridiculous for everyone to immediately understand the intention of the clue. Now everyone is synched that Donald knows their card is exactly red 3 (*except Emily, until r3 is discarded and she understands what has occurred, ofc). That was not true before, and we never are supposed to assume Asymmetric Information. 

If Alice is giving this clue in an attempt to Chop Move Emily, imo she's just making a clear mistake.

Once Emily's duplicate r3 discards, it seems like a much more simple, straightforward, intuitive, and useful convention to let her simply understand that it's **extremely likely** that this chain of events has, indeed, led to "more positive information about previously-clued cards". FWIW I will never CM from that clue in Emily's seat, barring very specific contex

**Comment by Zamiell** [comment]
> To me it just seems ridiculous for everyone to not immediately understand the intention of the clue. Now everyone is synched that Donald knows their card is exactly red 3

what if Donald already knew that the card was r3?
if that was true, then the actual point of Alice's clue would be to chop move Emily
but what if Donald only knew that the red card was r3 because he sees r5 and both copies of r4, and that isn't symmetric?

it is cognitive overload to consider all of this stuff and most of the people in the group are not good enough at the game to navigate this without making a mistake.
that's why the opinion of the doc is to KISS and simply never consider non-clue-receiver superpositions.

**Comment by pianoblook** [comment]
If we're taking into account every single potential superposition or not, then why would should anyone ever assume it **IS** Necessary, even in the example I gave (Cathy's hand being r4 x x [r]4?

For example:
> all 2s are down
> Cathy's hand is r4 x x [r]4
> Cathy can see r5 and both copies of r3 in Bob and Donald's hand
> Alice clues red and Bob Discharges y3

- Since Cathy already *knew*  her slot 4 was was red 4, this clue does not provide any additional information...**but only from her perspective**. 

So should she CM?

I would say absolutely not, because as far as the rest of the team is concerned this has given positive information. Assuming that she already knew her card exactly would be assuming a large amount of Assymetric Information, which we almost never do

**Comment by pianoblook** [comment]
> it is cognitive overload to consider all of this stuff and most of the people in the group are not good enough at the game to navigate this without making a mistake.

I 100% agree! Considering every superposition seems absurd, and I will basically never do that myself.

Therefore, I think we should expand UTDs to not be considered unnecessary when they can be seen to **very likely** provide extra positive info to previously clued cards!

Because surely we shouldn't be assuming everyone is tracking every possible superposition, right? :D

**Comment by Zamiell** [comment]
> I would say absolutely not, because as far as the rest of the team is concerned this has given positive information.

i agree.
so let's drop the talk about superpositions and focus on whether or not cards gain explicit positive information.

> I think we should expand UTDs to not be considered unnecessary when they can be seen to very likely provide extra positive info to previously clued cards!

that appears to be the way that the doc is written already, so you don't have to propose anything:

> A move is not unnecessary if it also added more positive information to other previously-clued cards in the hand of the person who received the clue.

**Comment by Zamiell** [comment]
in this context, the word "positive" in the doc explicitly means "touched with a positive clue".

you appear to be using the word "positive" in a different way. may i suggest using different terminology to make more clear what you are proposing?

**Comment by Zamiell** [comment]
I'll rewrite this section to be more clear:

```
  - A move is **not** *unnecessary* if it also "filled-in" previously-clued cards in the hand of the person who received the clue.
    - For simplicity, only positive clues "count" as being *Necessary*. "Filling-in" a card with a negative doesn't count.
```

**Comment by Zamiell** [comment]
[edited above comments to complain about the term "very likely"]

**Comment by pianoblook** [comment]
>   - A move is **not** *unnecessary* if it also "filled-in" previously-clued cards in the hand of the person who received the clue.

Technically this still runs in to the same dilemma as the r4-x-x-[r]4 issue. The clue doesn't *actually* 'fill-in' the r4 more than it was, but it sounds like we're on the same page that Cathy should not CM since it universally/symmetrically promises her slot 4 card to now be r4.

How about something like this: 

> A move is not Unnecessary if it leads to one or more previously-clued cards becoming globally known, whether through direct positive information or _Elimination Notes_.
- For simplicity, we don't count cards that happen to become known through negative information.


If we're agreeing that it seems overly complicated to force players to consider all possible superpositions and whatnot in the clue receiver's determination of whether it's Necessary, then I still strongly believe that this should still work regardless of whose cards are being touched

**Comment by Zamiell** [comment]
> The clue doesn't actually 'fill-in' the r4 more than it was

let's converge on using the same terminology:
1) yes, the clue does actually/literally "fills in" the card more than it was
2) it does **not** reduce the amount of entires inside of the superposition

**Comment by Zamiell** [comment]
> A move is not Unnecessary if it leads to one or more previously-clued cards becoming globally known, whether through direct positive information or Elimination Notes.

i think i found a problem with this definition:

- this is a rainbow game
- all 2s are down
- Cathy's hand is r4 x x [r]4
- Alice clues red and Bob Discharges y3

in this example, the superposition goes from [red 3, red 4, red 5, rainbow 3, rainbow 4, rainbow 5] --> [red 4, rainbow 4]
Thus, it was useful enough to be a necessary move (presumably you agree).
However, it does not result in cathy's card becoming globally known

**Comment by Zamiell** [comment]
that objection aside, i (still) dont like the definition, because it isn't globally known whether or not a card becomes globally known.
this is the crux of the issue.
the physical is 100% synchronous, and notes are not.
so let's create conventions based on the physical in order to avoid desync.

**Comment by pianoblook** [comment]
Thus, it was useful enough to be a necessary move (presumably you agree).
> However, it does not result in cathy's card becoming globally known

Well again, it will but only after the discard. That doesn't mean she should Chop Move immediately before discarding: the team can (and should, imo) clearly understand that the intention of this clue is for r4 to *become* globally known, but only after she discards

**Comment by pianoblook** [comment]
> in other words, it is not globally known when cards become globally known (if you are using notes to represent that).
> this is the crux of the issue.
> the physical is 100% synchronous, and notes are not.

I'm sorry, I'm still struggling to understand what you mean - or at least the significance of this distinction. 😭 

- We seem to agree that in the example we keep using, Cathy's card *will* be known to her as soon as she discards. Likewise, the team can all easily understand that it will become known when this happens.
- Even though she does not *currently* know the identity - and sure, therefore in that exact moment it still exists in a superposition - it will become known.
- This same "not quite yet but incredibly soon" impending collapse of superposition can be globally understood, regardless of who it is holding whatever trash card.

I feel like I must be missing something here, and I'm sorry if we're just talking past each other in some way

**Comment by pianoblook** [comment]
would it be helpful to use a different phrase, like "globally understood?"

**Comment by Zamiell** [comment]
> Well again, it will become globally known but only after she discards. That doesn't mean she should Chop Move immediately before discarding: the team can (and should, imo) clearly understand that the intention of this clue is for r4 to become globally known, but only after she discards

seems a bit too murky for my liking =(

**Comment by pianoblook** [comment]
Wait, so you're saying Cathy *SHOULD* chop move from that clue?

**Comment by pianoblook** [comment]
All I'm describing is how the convention currently holds up: we agree that she will have a globally known r4, but *only retroactively, once she discards* - hell, she might not discard for a few rounds, and thus the temporary superposition will continue until she finally discards it.

Which is all to say, since we're comfortable with *that* being "globally understood" (or w/e terminology we want to use to describe the team's internal thought processes), I see zero reason why we should exclude **other** situations that involve 'theoretically slightly desynchronized superpositions'

**Comment by Zamiell** [comment]
> Wait, so you're saying Cathy SHOULD chop move from that clue?

no.
i am saying that she should not chop move from that clue.
but the reason that she should not chop move is because it filled in a card.
(as opposed to because it "led to one or more previously-clued cards becoming globally known")

**Comment by Zamiell** [comment]
> All I'm describing is how the convention currently holds up: we agree that she will have a globally known r4, but only retroactively, once she discards - hell, she might not discard for a few rounds, and thus the temporary superposition will continue until she finally discards it.

indeed.
i agree that your definition is "more optimal" in the sense that it gets more games finished with a perfect score, if everybody plays perfectly.
however, the win-rate difference between your definition and the status quo definition seems low.
i do not think that this marginal benefit is outweighed by the added complexity of players having to retroactively account for things.
retroactive conventions have been historically shown to be error prone and destructive.

**Comment by Zamiell** [comment]
> We seem to agree that in the example we keep using, Cathy's card will be known to her as soon as she discards. Likewise, the team can all easily understand that it will become known when this happens.

no, this is not globally known.
for some players, Cathy's card is already known.
for other players, Cathy's card will become known only after Cathy discards.

**Comment by pianoblook** [comment]
Well I truly don't see the difference unfortunately, so I'm going to give up on this exchange, haha.

I guess the last thing I'll say is that if I ever perform the kind of UTD being proposed - the kind in which I clue someone *else's* duped card - I will **never** assume that the clued player (Emily, from your example way earlier) will Chop Move because from her perspective, I doubt she'll be considering every single superposition of everyone else's superpositions of Cathy's superpositions. Instead, once she discards the red card I expect her to think "hah cool, that was r3 so now kimbi knows her r3! very clever clue, Alice"

**Comment by Zamiell** [comment]
that thought process works great in the example where kim discovers that her card is r3, but the crux of the issue is that it doesn't generalize well to all game states
